be/src/storage/lake/compaction_scheduler.cpp

```diff
@@ -223,21 +231,72 @@ void CompactionScheduler::thread_task(int id) {
     }
 }
 
+static bool compaction_should_cancel(CompactionTaskContext* context) {
+    if (context->callback->has_error() || context->callback->timeout_exceeded()) {
+        return true;
+    }
+
+#ifndef BE_TEST
```

Comment:
make the `lake_compaction_check_valid_interval_minutes = 0` in be_test.conf to disable the check.

And possibly UT is required for this new added function.

Reply:
ut not possible, need fe

Reply:
that's not a good excuse to bypass UT for a large portion of code.